### ISSUE TITLE
Cant set default "Content-Type" value to request headers

### ISSUE THREAD

**Issue by kazuya32** [axios/axios#4339]
#### Describe the bug
Cant set default "Content-Type" value to request headers when config.data is object or URLSearchParams.

#### To Reproduce
Please run this on https://npm.runkit.com/axios.

```js
const axios = require("axios")

const axiosInstance = axios.create({ baseURL: "https://httpbin.org" });
axiosInstance.defaults.headers.common = { "Content-Type": "application/json;charset=UTF-8" };
const res = await axiosInstance.post("/post", { key: "value" });

console.log(res.request._header);
```

#### Expected behavior
If default headers have `Content-Type` value, the value should not be assumed as `application/json` or else.
In the above case, I expect request header's `Content-Type` value is `application/json;charset=UTF-8` but not because of [these useful code](https://github.com/axios/axios/blob/master/lib/defaults.js#L75).
In other words, at that case, [this condition](https://github.com/axios/axios/blob/master/lib/defaults.js#L12) should be false in `setContentTypeIfUnset` function.

#### Environment
Axios Version 0.24.0

#### Additional context/Screenshots
I think [Custom instance defaults](https://github.com/axios/axios#config-defaults) is commonly used on many codes so this issue is a high priority.

**Comment by marcuspoehls** [comment]
I’m running into the same issue: axios removes my `Content-Type: application/json` header from a custom instance.

My code works fine when using `Content-Type: application/x-www-form-urlencoded`.

**Comment by RadicalRoy** [comment]
This bug is preventing me from upgrading to 0.26.1. 

Can't set `Content-Type: application/json` on `GET` request in `react-native 0.63.4`. It works with our older version.

**Comment by jasonsaayman** [comment]
Hi 👋 

Please could you retry with the latest pre-release version and open a new issue should this error still be relevant?

Thanks
